server/views/partials/js-head.njk

```diff
@@ -1,30 +1,63 @@
+      if(b=-1!=f&&-1!=g||-1!=f&&-1!=h||-1!=g&&-1!=h)(b=f!=g&&f!=h&&g!=h)||(null===B&&(b=/AppleWebKit\/([0-9]+)(?:\.([0-9]+))/.exec(window.navigator.userAgent),B=!!b&&(536>parseInt(b[1],10)||536===parseInt(b[1],10)&&11>=parseInt(b[2],10))),b=B&&(f==u&&g==u&&h==u||f==v&&g==v&&h==v||f==w&&g==w&&h==w)),b=!b;b&&(null!==d.parentNode&&d.parentNode.removeChild(d),clearTimeout(q),a(c))}function G(){if((new Date).getTime()-F>=C)null!==d.parentNode&&d.parentNode.removeChild(d),b(c);else{var a=document.hidden;if(!0===
+      a||void 0===a)f=e.a.offsetWidth,g=n.a.offsetWidth,h=p.a.offsetWidth,t();q=setTimeout(G,50)}}var e=new r(k),n=new r(k),p=new r(k),f=-1,g=-1,h=-1,u=-1,v=-1,w=-1,d=document.createElement("div");d.dir="ltr";x(e,I(c,"sans-serif"));x(n,I(c,"serif"));x(p,I(c,"monospace"));d.appendChild(e.a);d.appendChild(n.a);d.appendChild(p.a);document.body.appendChild(d);u=e.a.offsetWidth;v=n.a.offsetWidth;w=p.a.offsetWidth;G();z(e,function(a){f=a;t()});x(e,I(c,'"'+c.family+'",sans-serif'));z(n,function(a){g=a;t()});x(n,
+      I(c,'"'+c.family+'",serif'));z(p,function(a){h=a;t()});x(p,I(c,'"'+c.family+'",monospace'))})})};"undefined"!==typeof module?module.exports=A:(window.FontFaceObserver=A,window.FontFaceObserver.prototype.load=A.prototype.load);}());
+
+      var rootElement = document.documentElement;
+      var enhancedClass = 'enhanced';
+      var enhancedScriptPath = '/assets/js/app.js';
+
+      var WBSubset = new FontFaceObserver('Wellcome Bold Web Subset', {weight: 'bold'});
+      var WB = new FontFaceObserver('Wellcome Bold Web', {weight: 'bold'});
+      var HNL = new FontFaceObserver('Helvetica Neue Light Web');
+      var HNM = new FontFaceObserver('Helvetica Neue Medium Web');
+      var LR = new FontFaceObserver('Lettera Regular Web');
+
+      function addEnhancedClass() {
+        rootElement.classList.add(enhancedClass);
+      }
+
+      function removeEnhancedClass() {
+        rootElement.classList.remove(enhancedClass);
+      }
+
+      function loadJS(src) {
+        var ref = window.document.getElementsByTagName('script')[ 0 ];
+        var script = window.document.createElement('script');
+        script.src = src;
+        script.async = true;
+        ref.parentNode.insertBefore(script, ref);
+        return script;
+      };
+
+      if (sessionStorage.fontsLoaded) {
+        rootElement.classList.add('font-stage-1','font-stage-2');
+      } else {
+        WBSubset.load().then(function() {
+          rootElement.classList.add('font-stage-1');
+          Promise.all([WB.load(), HNL.load(), HNM.load(), LR.load()]).then(() => {
+            rootElement.classList.add('font-stage-2');
+            sessionStorage.fontsLoaded = true;
+          });
         });
-      });
+      }
+
+      addEnhancedClass();
+
+      var script = loadJS(enhancedScriptPath);
+      var fallback = setTimeout(removeEnhancedClass, 8000);
```

Comment:
We add it straight away, to prevent jump from non enhanced to enhanced styling, when javascript loads. So if the javascript fails to load, it's then removed.